packages/js-drive/lib/identity/masternode/handleNewMasternodeFactory.js

```diff
@@ -0,0 +1,94 @@
+const Identifier = require('@dashevo/dpp/lib/identifier/Identifier');
+const { hash } = require('@dashevo/dpp/lib/util/hash');
+const IdentityPublicKey = require('@dashevo/dpp/lib/identity/IdentityPublicKey');
+const Transaction = require('@dashevo/dashcore-lib/lib/transaction');
+
+/**
+ *
+ * @param {DashPlatformProtocol} transactionalDpp
+ * @param {DriveStateRepository|CachedStateRepositoryDecorator} stateRepository
+ * @param {createMasternodeIdentity} createMasternodeIdentity
+ * @param {DataContractStoreRepository} dataContractRepository
+ * @param {Identifier} masternodeRewardSharesContractId
+ * @return {handleNewMasternode}
+ */
+function handleNewMasternodeFactory(
+  transactionalDpp,
+  stateRepository,
+  createMasternodeIdentity,
+  dataContractRepository,
+  masternodeRewardSharesContractId,
+) {
+  /**
+   * @typedef handleNewMasternode
+   * @param {SimplifiedMNListEntry} masternodeEntry
+   * @return {Promise<{
+   *            create: Document[],
+   *            delete: Document[],
+   * }>}
+   */
+  async function handleNewMasternode(masternodeEntry) {
+    const rawTransaction = await stateRepository
+      .fetchTransaction(masternodeEntry.proRegTxHash);
+
+    const { extraPayload: proRegTxPayload } = new Transaction(rawTransaction.data);
+
+    // Create a masternode identity
+    const masternodeIdentityId = Identifier.from(
+      hash(
+        Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+      ),
+    );
+
+    await createMasternodeIdentity(
+      masternodeIdentityId,
+      Buffer.from(proRegTxPayload.keyIDOwner, 'hex'),
+      IdentityPublicKey.TYPES.ECDSA_HASH160,
+    );
+
+    const documentsToCreate = [];
+    const documentsToDelete = [];
+
```

Comment:
I would add a comment that we need to crate reward shares only if it's enabled in proRegTx